src/nspawn/nspawn-mount.c

```diff
@@ -83,13 +83,22 @@ static int custom_mount_compare(const CustomMount *a, const CustomMount *b) {
         return CMP(a->type, b->type);
 }
 
-static bool source_path_is_valid(const char *p) {
+static int source_path_parse(const char *p, bool allow_empty, char **ret) {
         assert(p);
+        assert(ret);
+
+        if (isempty(p)) {
+                if (!allow_empty)
+                        return -EINVAL;
 
-        if (*p == '+')
-                p++;
+                *ret = NULL;
+                return 0;
+        }
+
+        if (*p == '+' && !path_is_absolute(p++))
```

Comment:
i seriously dislike if checks with side-effects... it's confusing, so confusing in fact that this is wrong, no? you mean ++p, not p++...

Reply:
Yeah,  I figured this was pushing it a bit, I split it into multiple lines